components/setting_item_max.jsx

```diff
@@ -132,7 +132,7 @@ export default class SettingItemMax extends React.PureComponent {
         if (this.props.shiftEnter && e.keyCode === Constants.KeyCodes.ENTER && e.shiftKey) {
             return;
         }
-        if (isKeyPressed(e, Constants.KeyCodes.ENTER) && this.props.submit && e.target.tagName !== 'SELECT') {
+        if (isKeyPressed(e, Constants.KeyCodes.ENTER) && this.props.submit && e.target.tagName !== 'SELECT' && e.target.parentElement && e.target.parentElement.className !== 'react-select__input') {
```

Comment:
I don't really like the look of this since it seems like we're just trying to duplicate a form's submit event in a more fragile way, but since this component is used in so many places, I'd be worried about breaking it by trying to change it now

Reply:
Cool, we basically just do not want to submit on enter, we can hopefully add a prop and then use it, that just tells the settings to submit on enter or not.

Reply:
I just tested a bit locally, and react-select does behave correctly in a form, so that might be something we consider fixing in the future